File: 0011-container-with-most-water/0011-container-with-most-water.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var maximum = 0
        
        var front = 0
        var back = height.count - 1
        
        while front < back {
            let bounds = (back - front) * min(height[front], height[back])
            maximum = max(maximum, bounds)
            
            if height[front] < height[back] {
                front += 1
            } else {
                back -= 1
            }
        }
        
        return maximum
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
    let result = 0;
    let left = 0;
    let right = height.length - 1;

    while (left < right) {
        const leftPillar = height[left];
        const rightPillar = height[right];
        const area = (right - left) * Math.min(leftPillar, rightPillar);

        result = Math.max(result, area);

        if (leftPillar > rightPillar) {
            right--;
        } else {
            left++;
        }
    }

    return result;
};